pkg/resource/record_set/sdk.go

```diff
@@ -0,0 +1,740 @@
+			}
+			if elem.AliasTarget.HostedZoneId != nil {
+				f0.HostedZoneID = elem.AliasTarget.HostedZoneId
+			}
+			ko.Spec.AliasTarget = f0
+		} else {
+			ko.Spec.AliasTarget = nil
+		}
+		if elem.CidrRoutingConfig != nil {
+			f1 := &svcapitypes.CIDRRoutingConfig{}
+			if elem.CidrRoutingConfig.CollectionId != nil {
+				f1.CollectionID = elem.CidrRoutingConfig.CollectionId
+			}
+			if elem.CidrRoutingConfig.LocationName != nil {
+				f1.LocationName = elem.CidrRoutingConfig.LocationName
+			}
+			ko.Spec.CIDRRoutingConfig = f1
+		} else {
+			ko.Spec.CIDRRoutingConfig = nil
+		}
+		if elem.Failover != nil {
+			ko.Spec.Failover = elem.Failover
+		} else {
+			ko.Spec.Failover = nil
+		}
+		if elem.GeoLocation != nil {
+			f3 := &svcapitypes.GeoLocation{}
+			if elem.GeoLocation.ContinentCode != nil {
+				f3.ContinentCode = elem.GeoLocation.ContinentCode
+			}
+			if elem.GeoLocation.CountryCode != nil {
+				f3.CountryCode = elem.GeoLocation.CountryCode
+			}
+			if elem.GeoLocation.SubdivisionCode != nil {
+				f3.SubdivisionCode = elem.GeoLocation.SubdivisionCode
+			}
+			ko.Spec.GeoLocation = f3
+		} else {
+			ko.Spec.GeoLocation = nil
+		}
+		if elem.HealthCheckId != nil {
+			ko.Spec.HealthCheckID = elem.HealthCheckId
+		} else {
+			ko.Spec.HealthCheckID = nil
+		}
+		if elem.MultiValueAnswer != nil {
+			ko.Spec.MultiValueAnswer = elem.MultiValueAnswer
+		} else {
+			ko.Spec.MultiValueAnswer = nil
+		}
+		if elem.Name != nil {
+			if ko.Spec.Name != nil {
+				if *elem.Name != *ko.Spec.Name {
+					continue
+				}
+			}
+			ko.Spec.Name = elem.Name
+		} else {
+			ko.Spec.Name = nil
+		}
+		if elem.Region != nil {
+			ko.Spec.Region = elem.Region
+		} else {
+			ko.Spec.Region = nil
+		}
+		if elem.ResourceRecords != nil {
+			f8 := []*svcapitypes.ResourceRecord{}
+			for _, f8iter := range elem.ResourceRecords {
+				f8elem := &svcapitypes.ResourceRecord{}
+				if f8iter.Value != nil {
+					f8elem.Value = f8iter.Value
+				}
+				f8 = append(f8, f8elem)
+			}
+			ko.Spec.ResourceRecords = f8
+		} else {
+			ko.Spec.ResourceRecords = nil
+		}
+		if elem.SetIdentifier != nil {
+			if ko.Spec.SetIdentifier != nil {
+				if *elem.SetIdentifier != *ko.Spec.SetIdentifier {
+					continue
+				}
+			}
+			ko.Spec.SetIdentifier = elem.SetIdentifier
+		} else {
+			ko.Spec.SetIdentifier = nil
+		}
+		if elem.TTL != nil {
+			ko.Spec.TTL = elem.TTL
+		} else {
+			ko.Spec.TTL = nil
+		}
+		if elem.Weight != nil {
+			ko.Spec.Weight = elem.Weight
+		} else {
+			ko.Spec.Weight = nil
+		}
+		found = true
+		break
+	}
+	if !found {
+		return nil, ackerr.NotFound
+	}
+
+	rm.setStatusDefaults(ko)
+
+	// Status represents whether record changes have been fully propagated to all
+	// Route 53 authoritative DNS servers. The current status for the propagation
+	// should be updated per reconciliation
+	if ko.Status.Status != nil {
+		err = rm.syncStatus(ctx, ko)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
How about making this part of the `sdkFind`, anyways..

```suggestion
		err = rm.syncStatus(ctx, ko)
		if err != nil {
			return nil, err
		}
		// Also force a requeue if the Change is still pending
		if sync.Status.Status != *"PENDING" {
		    ackcondition.SetSynced(&resource{ko}, corev1.ConditionFalse, nil, nil)
		}
```